### PR TITLE
Add ASGI module to Ariadne

### PR DESCRIPTION
**Pull request by patrys**
Work in progress implementation of an ASGI server based on Starlette.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,9 +93,14 @@ def resolve_person_fullname(person, *_):
 
 # Create executable GraphQL schema
 schema = make_executable_schema(type_defs, [query, person])
+# Create an ASGI app
```
Comment (rafalp) [inline README.md]:
```suggestion

# Create an ASGI app using the schema
```

File: ariadne/asgi.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+GQL_STOP = "stop"  # Client -> Server
+
+
+class GraphQL:
+    def __init__(self, schema: GraphQLSchema, *, keepalive: float = None):
+        self.keepalive = keepalive
+        self.schema = schema
+
+    def __call__(self, scope: Scope):
+        assert scope["type"] in {"http", "websocket"}
```
Comment (rafalp) [inline ariadne/asgi.py]: We should remove either assert or `ValueError` from end of this method

File: ariadne/asgi.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+                payload["errors"] = [format_error(e) for e in result.errors]
+            await self.send_json(
+                websocket, {"type": GQL_DATA, "id": operation_id, "payload": payload}
+            )
+        await self.send_json(websocket, {"type": GQL_COMPLETE, "id": operation_id})
+
+    async def keep_connection_alive(self, websocket: WebSocket):
+        if not self.keepalive:
+            return
+        while True:
```
Comment (rafalp) [inline ariadne/asgi.py]:
```suggestion
        while websocket.application_state != WebSocketState.DISCONNECTED::
```

We can drop bottom `if/break` that way

File: ariadne/asgi.py [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+    async def send_json(self, websocket, message) -> None:
+        message = json.dumps(message)
+        await websocket.send_text(message)
+
+    async def graphql_ws_server(self, websocket: WebSocket) -> None:
+        subscriptions: Dict[str, AsyncGenerator] = {}
+        await websocket.accept("graphql-ws")
+        try:
+            while True:
```
Comment (rafalp) [inline ariadne/asgi.py]: Could we break this block down into smaller methods?

File: docs/modularization.rst [hunk 5]
```diff
@@ -28,8 +29,9 @@ To load schema from file or directory, you can use the ``load_schema_from_path``
     # build an executable schema
     schema = make_executable_schema(type_defs)
 
-    # Start server that can't execute any queries, but allows you to browse your schema
-    start_simple_server(schema)
+    app = GraphQL(schema)
```
Comment (rafalp) [inline docs/modularization.rst]: Move this below next two comments and reword those to something like "Use schema to construct GraphQL app for an ASGI server"?

**Comment by patrys** [comment]
Checks currently fail because `black` is configured to check generated files, @rafalp will take care of that.

**Pull request merged**